bundles/org.openhab.binding.mideaac/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,394 @@
+				<context>email</context>
+				<label>Email</label>
+				<description>Email for cloud account chosen in Cloud Provider.</description>
+				<default>nethome+us@mailinator.com</default>
+			</parameter>
+			<parameter name="password" type="text" required="false">
+				<context>password</context>
+				<label>Password</label>
+				<description>Password for cloud account chosen in Cloud Provider.</description>
+				<default>password1</default>
+			</parameter>
+			<parameter name="token" type="text" required="false">
+				<context>token</context>
+				<label>Token</label>
+				<description>Secret Token (length 128 HEX) used for secure connection authentication used with devices v3 (if not
+					known, enter email, password for Cloud account to retrieve it).</description>
+			</parameter>
+			<parameter name="key" type="text" required="false">
+				<context>key</context>
+				<label>Key</label>
+				<description>Secret Key (length 64 HEX) used for secure connection authentication used with devices v3 (if not
+					known, enter email, password for Cloud account to retrieve it).</description>
+			</parameter>
+			<parameter name="pollingTime" type="decimal" required="true" min="30" unit="s">
+				<context>pollingTime</context>
+				<label>Poll frequency in seconds</label>
+				<description>Poll device status. Minimum is 30, default 60.</description>
+				<default>60</default>
+			</parameter>
+			<parameter name="energyPoll" type="decimal" required="true" min="0" unit="m">
+				<context>energyPoll</context>
+				<label>Energy poll in minutes</label>
+				<description>Energy polling frequency. default 0 ; (in case not supported).</description>
+				<default>0</default>
+			</parameter>
+			<parameter name="keyTokenUpdate" type="decimal" required="false" min="0" unit="d">
+				<context>keyTokenUpdate</context>
+				<label>Key token update frequency in days</label>
+				<description>Update the Key and Token from the cloud, default 0 to disable.</description>
+				<default>0</default>
+			</parameter>
+			<parameter name="timeout" type="decimal" required="true" min="2" max="10" unit="s">
+				<context>timeout</context>
+				<label>Socket timeout in seconds</label>
+				<description>Connecting socket timeout. Minimum is 2, maximum is 10 (4 is default).</description>
+				<default>4</default>
+			</parameter>
+			<parameter name="promptTone" type="boolean" required="true">
+				<context>promptTone</context>
+				<label>Prompt tone</label>
+				<description>After sending a command device will play "ding" tone when command is received and executed.</description>
+				<default>false</default>
+			</parameter>
+			<parameter name="version" type="decimal" required="true">
+				<context>version</context>
+				<label>AC Version</label>
+				<description>Version 3 requires Token, Key and Cloud provider. Version 2 doesn't.</description>
+				<default>0</default>
+			</parameter>
+			<parameter name="energyDecode" type="boolean" required="true">
```

Comment:
Would it be possible to detect this while decoding? It would make this parameter obsolete.

Reply:
Not sure which one you are asking about. The version (which is decoded in discovery) or the energyDecode which is not. HA has a flag I had to try both methods to figure out which one.